HealthcareSystem/PatientWebApp/Validators/SurveyValidator.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Backend.Service;
+using PatientWebApp.DTOs;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientWebApp.Validators
+{
+    public class SurveyValidator
+    {
+        private readonly ISurveyService _surveyService;
+
+        public SurveyValidator(ISurveyService surveyService)
+        {
+            _surveyService = surveyService;
+        }
+
+        public void ValidateSurveyFields(SurveyDTO surveyDTO)
+        {
+            if (surveyDTO == null)
+                throw new ValidationException("Survey cannot be null.");
+            if (surveyDTO.DoctorJmbg == null || surveyDTO.DoctorJmbg == "")
+                throw new ValidationException("The doctor does not exist!");
+            if (!IsValidGrade(surveyDTO.BehaviorOfDoctor))
+                throw new ValidationException("The rating is out of range.");
+            if (!IsValidGrade(surveyDTO.DoctorProfessionalism))
+                throw new ValidationException("The rating is out of range.");
+            if (!IsValidGrade(surveyDTO.GettingAdviceByDoctor))
+                throw new ValidationException("The rating is out of range.");
+            if (!IsValidGrade(surveyDTO.AvailabilityOfDoctor))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.BehaviorOfMedicalStaff))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.MedicalStaffProfessionalism))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.GettingAdviceByMedicalStaff))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.EaseInObtainingFollowupInformationAndCare))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.Nursing))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.Cleanliness))
+                throw new ValidationException("The rating is out of range."); 
+            if (!IsValidGrade(surveyDTO.OverallRating))
+                throw new ValidationException("The rating is out of range.");
+            if (!IsValidGrade(surveyDTO.SatisfiedWithDrugAndInstrument))
+                throw new ValidationException("The rating is out of range.");
+        }
```

Comment:
This method is too big, but if we make some structure for some of the properties in DTO, we then can validate those of structure and in them some of the properties. I am not saying I am right, I just want to talk about that idea, what you think @jelenazeljko55 ?